### PR TITLE
Changed token classification format to IOB

### LINKED ISSUE #2030: [FEAT] Support simpler IOB annotation for token classification
**Is your feature request related to a problem? Please describe.**
In addition to the existing support for token classification with the `(class, start_span, end_span)` label annotation format, support the IOB format of `['ttoken1', 'token2'] ['class1', 'class2']` (which doesn't allow for mismatch of tokenization between the labels and the model predictions.

**Describe the solution you'd like**
Enable the user to provide the simpler IOB annotation as input for the TextData

### PR DESCRIPTION
**Pull request by nirhutnik**
closes #2030

### CODE REVIEW

File: deepchecks/nlp/metric_utils/token_classification.py [hunk 1]
```diff
@@ -70,172 +65,15 @@ def get_scorer_dict(span_aligner: TSpanAligner,
...
-            # A list of the spans, sorted by their starting position
-            label_spans = [(annotation[1], annotation[2]) for annotation in sample_y_true]
-            pred_spans = [(annotation[1], annotation[2]) for annotation in sample_y_pred]
-
-            # Here we want to find the prediction spans that match specific label spans, and discard the rest.
-            # We want to do that in O(N*log(N)) - assuming both lists are of length N approximately. Thus, the use of
-            # np.searchsorted
-            if len(sample_y_true) > 0:
-                # Indices of where to place elements of label_spans in pred_spans so that pred_spans is still sorted.
-                sorted_label_in_pred_index = np.searchsorted(list(map(lambda x: x[0], pred_spans)),
```
Comment (noamzbr) [inline deepchecks/nlp/metric_utils/token_classification.py:181]: RIP

Comment (noamzbr) [reply deepchecks/nlp/metric_utils/token_classification.py:181]: Is this kept somewhere else? A gist or something?

File: deepchecks/nlp/text_data.py [hunk 2]
```diff
@@ -39,8 +39,11 @@ class TextData:
 
     Parameters
     ----------
-    raw_text : t.Sequence[str]
+    raw_text : t.Sequence[str], default: None
         The raw text data, a sequence of strings representing the raw text of each sample.
+    tokenized_text : t.Sequence[t.Sequence[str]], default: None
+        The tokenized text data, a sequence of sequences of strings representing the tokenized text of each sample.
+        Only relevant for task_type 'token_classification'.
```
Comment (noamzbr) [inline deepchecks/nlp/text_data.py:46]: Maybe say that raw text will be generated by ' '.join-ing this?

File: deepchecks/nlp/text_data.py [hunk 3]
```diff
@@ -122,15 +138,26 @@ def _validate_text(raw_text: t.Sequence[str]):
...
-    def _validate_and_set_label(self, label: t.Optional[TTextLabel], raw_text: t.Sequence[str]):
+    @staticmethod
+    def _validate_tokenized_text(tokenized_text: t.Sequence[t.Sequence[str]]):
+        """Validate tokenized text format."""
+        if not isinstance(tokenized_text, collections.abc.Sequence):
+            raise DeepchecksValueError('tokenized_text must be a sequence')
+        if not all(isinstance(x, collections.abc.Sequence) for x in tokenized_text):
+            raise DeepchecksValueError('tokenized_text must be a Sequence of sequences')
+        if not all(isinstance(x, str) for tokens in tokenized_text for x in tokens):
+            raise DeepchecksValueError('tokenized_text must be a Sequence of sequences of strings')
```
Comment (noamzbr) [inline deepchecks/nlp/text_data.py:149]: Would just always print this last error

File: tests/nlp/nlp_conftest.py [hunk 4]
```diff
@@ -83,7 +84,20 @@ def movie_reviews_data_negative():
...
                      'How much wood can a wood chuck chuck?'],
-                    [[('B-PER', 0, 4)], [('B-PER', 0, 4), ('B-GEO', 14, 20), ('B-GEO', 25, 30)], []],
+                    label=[['B-PER', 'O', 'O', 'O', 'O'], ['B-PER', 'O', 'O', 'B-GEO', 'O', 'B-GEO'],
+                     ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O']],
                     task_type='token_classification')
+
+@pytest.fixture(scope='session')
+def wikiann():
+    """Wikiann dataset for token classification"""
+    dataset = load_dataset('wikiann', name='en', split='train')
```
Comment (noamzbr) [inline tests/nlp/nlp_conftest.py:96]: Can we load less data?

**Review by noamzbr — APPROVED** [review]

File: deepchecks/nlp/metric_utils/token_classification.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/nlp/metric_utils/token_classification.py:181]: I have it locally, thought of uploading to drive. Other ideas?

File: deepchecks/nlp/text_data.py (hunk 2 above)
Comment (nirhutnik) [reply deepchecks/nlp/text_data.py:46]: Done

File: deepchecks/nlp/text_data.py (hunk 3 above)
Comment (nirhutnik) [reply deepchecks/nlp/text_data.py:149]: Done

File: tests/nlp/nlp_conftest.py (hunk 4 above)
Comment (nirhutnik) [reply tests/nlp/nlp_conftest.py:96]: It loads 10,000 short sentences. Not a lot IMO

**Pull request merged**